lib/datadog/di.rb

```diff
@@ -58,6 +59,32 @@ def activate_tracking!
         (@code_tracker ||= CodeTracker.new).start
       end
 
+      # Activates code tracking if possible.
+      #
+      # This method does nothing if invoked in an environment that does not
+      # implement required trace points for code tracking (MRI Ruby < 2.6,
+      # JRuby) and rescues any exceptions that may be raised by downstream
+      # DI code.
+      def activate_tracking
+        # :script_compiled trace point was added in Ruby 2.6.
+        if RUBY_VERSION >= '2.6'
+          begin
+            # Activate code tracking by default because line trace points will not work
+            # without it.
+            Datadog::DI.activate_tracking!
+          rescue => exc
+            if defined?(Datadog.logger)
+              Datadog.logger.warn("Failed to activate code tracking for DI: #{exc.class}: #{exc}")
+            else
+              # We do not have Datadog logger potentially because DI code tracker is
+              # being loaded early in application boot process and the rest of datadog
+              # wasn't loaded yet. Output to standard error.
+              warn("Failed to activate code tracking for DI: #{exc.class}: #{exc}")
+            end
+          end
```

Comment:
Small nit:
```suggestion
        return if RUBY_VERSION >= '2.6'

        # Activate code tracking by default because line trace points will not work
        # without it.
        Datadog::DI.activate_tracking!
        rescue => exc
          if defined?(Datadog.logger)
            Datadog.logger.warn("Failed to activate code tracking for DI: #{exc.class}: #{exc}")
          else
            # We do not have Datadog logger potentially because DI code tracker is
            # being loaded early in application boot process and the rest of datadog
            # wasn't loaded yet. Output to standard error.
            warn("Failed to activate code tracking for DI: #{exc.class}: #{exc}")
        end
```

Reply:
I think you meant to reverse the sign on comparison which I did.